src/core/nixl_agent.cpp

```diff
@@ -81,6 +82,9 @@ nixlAgentData::nixlAgentData(const std::string &name,
         throw std::invalid_argument("Agent needs a name");
 
     memorySection = new nixlLocalSection();
+
+    const char *telemetry = std::getenv("NIXL_EN_TELEMETRY");
```

Comment:
minor: names seems a bit strange
i'd call it something like `NIXL_TELEMETRY_ENABLE` or similar